### PR TITLE
Add ability to catch entities with fishing

### PR DESCRIPTION
**Pull request by ErikSzabo**
This PR adds a new entity loot type for fishing (or it can be added to other loot tables as well, it just didn't really make sense).
The caught entity can be a vanilla entity (typically a mob) or a MythicMobs using the `mythicmobs:` prefix. Entity health, damage, and level can also be configured.

The vertical and horizontal velocities are also configurable for fishing but already have default values.

Example configuration in `fishing.yml`:

```yml
type: fishing
pools:
  rare:
    base_chance: 100
    selection_priority: 1
    loot:
      - type: entity
        entity: zombie
        name: "&6My Little Zombie"
        weight: 50
        health: 100
        damage: 100
        velocity:
          horizontal: 1.2
          vertical: 1.3
      - type: entity
        entity: mythicmobs:SkeletonKing
        weight: 50
        level: 10
```

### CODE REVIEW

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntityProperties.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package dev.aurelium.auraskills.bukkit.loot.entity;
+
+import org.spongepowered.configurate.ConfigurationNode;
+
+public record EntityProperties(String entityId,
+                               String name,
+                               Integer level,
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntityProperties.java:7]: These should be primitives not the class versions

Comment (Archy-X) [reply bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntityProperties.java:7]: Ah nevermind I see you null check them later

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/hooks/mythicmobs/loot/MythicEntitySupplier.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import dev.aurelium.auraskills.bukkit.AuraSkills;
+import dev.aurelium.auraskills.bukkit.loot.entity.EntityProperties;
+import dev.aurelium.auraskills.bukkit.loot.entity.EntitySupplier;
+import io.lumine.mythic.bukkit.BukkitAdapter;
+import io.lumine.mythic.bukkit.MythicBukkit;
+import io.lumine.mythic.core.mobs.ActiveMob;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+
+public class MythicEntitySupplier extends EntitySupplier {
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/hooks/mythicmobs/loot/MythicEntitySupplier.java:12]: Add empty line between class definition and constructor

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntityProperties.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                               Float horizontalVelocity,
+                               Float verticalVelocity) {
+
+    public static EntityProperties fromConfig(ConfigurationNode config) {
+        String[] id = config.node("entity").getString().split(":");
+
+        return new EntityProperties(
+                id.length > 1 ? id[1] : id[0],
+                config.node("name").getString(),
+                config.node("level").empty() ? null :config.node("level").getInt(),
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntityProperties.java:19]: Ensure space after colon

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntitySupplier.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package dev.aurelium.auraskills.bukkit.loot.entity;
+
+import dev.aurelium.auraskills.bukkit.AuraSkills;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+
+public abstract class EntitySupplier {
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntitySupplier.java:7]: Empty line after

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntitySupplier.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    private final EntityProperties entityProperties;
+
+    public EntitySupplier(EntityProperties entityProperties) {
+        this.entityProperties = entityProperties;
+    }
+
+    public abstract Entity spawnEntity(AuraSkills plugin, Location location);
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/EntitySupplier.java:14]: Add empty line between methods

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/VanillaEntityParser.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package dev.aurelium.auraskills.bukkit.loot.entity;
+
+import dev.aurelium.auraskills.bukkit.loot.parser.CustomEntityParser;
+import org.spongepowered.configurate.ConfigurationNode;
+
+public class VanillaEntityParser implements CustomEntityParser {
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/VanillaEntityParser.java:6]: Add empty line after

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/handler/LootHandler.java [hunk 7]
```diff
@@ -116,6 +120,36 @@ protected void giveFishingItemLoot(Player player, ItemLoot loot, PlayerFishEvent
...
+
+        if (dropEvent.isCancelled()) {
+            loot.getEntity().removeEntity(entity);
+            return;
+        }
+
+        itemEntity.setItemStack(new ItemStack(Material.AIR));
+
+        Float hVelocity = loot.getEntity().getEntityProperties().horizontalVelocity();
+        if(hVelocity == null) hVelocity = 1.2f;
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/handler/LootHandler.java:140]: Space after if

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/parser/EntityLootParser.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import dev.aurelium.auraskills.api.loot.LootParsingContext;
+import dev.aurelium.auraskills.bukkit.loot.LootManager;
+import dev.aurelium.auraskills.bukkit.loot.entity.EntitySupplier;
+import dev.aurelium.auraskills.bukkit.loot.type.EntityLoot;
+import dev.aurelium.auraskills.common.util.data.Validate;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+
+
+public class EntityLootParser implements LootParser {
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/parser/EntityLootParser.java:14]: Add empty line

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/type/EntityLoot.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package dev.aurelium.auraskills.bukkit.loot.type;
+
+import dev.aurelium.auraskills.api.loot.Loot;
+import dev.aurelium.auraskills.api.loot.LootValues;
+import dev.aurelium.auraskills.bukkit.loot.entity.EntitySupplier;
+
+public class EntityLoot extends Loot {
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/type/EntityLoot.java:7]: Add empty line

File: bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/VanillaEntitySupplier.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+
+        if (getEntityProperties().name() != null) {
+            entity.setCustomName(plugin.getMessageProvider().applyFormatting(getEntityProperties().name()));
+            entity.setCustomNameVisible(true);
+        }
+
+        if (getEntityProperties().level() != null) {
+            entity.setMetadata("aura_skills_level", new FixedMetadataValue(plugin, getEntityProperties().level()));
```
Comment (Archy-X) [inline bukkit/src/main/java/dev/aurelium/auraskills/bukkit/loot/entity/VanillaEntitySupplier.java:40]: auraskills should be one word without an underscore

**Review by Archy-X — CHANGES_REQUESTED** [review]
Initial formatting review, will need to test in game personally later.

**Comment by ErikSzabo** [comment]
Made those changes. Also enabled auto formatting in Intellij on save for the future :'D

**Review by Archy-X — APPROVED** [review]

**Pull request merged**